Sources/Swift/Protocol/Codable/SentryEventCodable.swift

```diff
@@ -18,39 +16,73 @@ import Foundation
+        self.timestamp = try container.decode(Date.self, forKey: .timestamp)
+        self.startTimestamp = try container.decodeIfPresent(Date.self, forKey: .startTimestamp)
+
+        if let rawLevel = try container.decodeIfPresent(String.self, forKey: .level) {
+            let level = SentryLevelHelper.levelForName(rawLevel)
+            SentryLevelBridge.setBreadcrumbLevelOn(self, level: level.rawValue)
+        } else {
+            SentryLevelBridge.setBreadcrumbLevelOn(self, level:
+            SentryLevel.none.rawValue)
+        }
+
+        self.platform = try container.decode(String.self, forKey: .platform)
+        self.logger = try container.decodeIfPresent(String.self, forKey: .logger)
+        self.serverName = try container.decodeIfPresent(String.self, forKey: .serverName)
+        self.releaseName = try container.decodeIfPresent(String.self, forKey: .releaseName)
+        self.dist = try container.decodeIfPresent(String.self, forKey: .dist)
+        self.environment = try container.decodeIfPresent(String.self, forKey: .environment)
+        self.transaction = try container.decodeIfPresent(String.self, forKey: .transaction)
+        self.type = try container.decodeIfPresent(String.self, forKey: .type)
+        self.tags = try container.decodeIfPresent([String: String].self, forKey: .tags)
+
+        self.extra = decodeArbitraryData {
+            try container.decodeIfPresent([String: ArbitraryData].self, forKey: .extra)
+        }
+        self.sdk = decodeArbitraryData {
+            try container.decodeIfPresent([String: ArbitraryData].self, forKey: .sdk)
+        }
+
+        self.modules = try container.decodeIfPresent([String: String].self, forKey: .modules)
+        self.fingerprint = try container.decodeIfPresent([String].self, forKey: .fingerprint)
+        self.user = try container.decodeIfPresent(UserDecodable.self, forKey: .user)
+        
+        self.context = decodeArbitraryData {
+            try container.decodeIfPresent([String: [String: ArbitraryData]].self, forKey: .context)
+        }
+
+        if let rawThreads = try container.decodeIfPresent([String: [SentryThreadDecodable]].self, forKey: .threads) {
+            self.threads = rawThreads["values"]
+        }
+            
+        if let rawExceptions = try container.decodeIfPresent([String: [ExceptionDecodable]].self, forKey: .exception) {
+            self.exceptions = rawExceptions["values"]
+        }
+        
+        self.stacktrace = try container.decodeIfPresent(SentryStacktraceDecodable.self, forKey: .stacktrace)
+        
+        if let rawDebugMeta = try container.decodeIfPresent([String: [DebugMetaDecodable]].self, forKey: .debugMeta) {
+            self.debugMeta = rawDebugMeta["images"]
+        }
+        
+        self.breadcrumbs = try container.decodeIfPresent([BreadcrumbDecodable].self, forKey: .breadcrumbs)
+        self.request = try container.decodeIfPresent(SentryRequestDecodable.self, forKey: .request)
```

Comment:
`m`: This code is now duplicate. Can't we create a common private init method or something else to ditch the duplication?